### PR TITLE
feat(auth): add delete user api keys mutation

### LINKED ISSUE #4059: [auth][graphql] `deleteUserAPIKeys` mutation
```graphql
mutation deleteUserAPIKeys(keyIds: [String!]!) {
  apiKeys {
    id
    name
  }
}
```

Only admins and the user who created a key should be able to delete it.

### PR DESCRIPTION
**Pull request by Parker-Stafford**
resolves #4059 

image

### CODE REVIEW

File: src/phoenix/server/api/mutations/api_key_mutations.py [hunk 1]
```diff
@@ -151,10 +151,21 @@ async def create_user_api_key(
...
+    ) -> DeleteApiKeyMutationPayload:
         assert (token_store := info.context.token_store) is not None
         api_key_id = from_global_id_with_expected_type(
             input.id, expected_type_name=SystemApiKey.__name__
         )
         await token_store.revoke(ApiKeyId(api_key_id))
-        return DeleteSystemApiKeyMutationPayload(id=input.id, query=Query())
+        return DeleteApiKeyMutationPayload(id=input.id, query=Query())
+
+    @strawberry.mutation(permission_classes=[HasSecret, IsAuthenticated])  # type: ignore
```
Comment (mikeldking) [inline src/phoenix/server/api/mutations/api_key_mutations.py:162]:
```suggestion
    @strawberry.mutation(permission_classes=[HasSecret, IsAuthenticated, IsAdmin])  # type: ignore
```
Need to be an admin to delete a user's key

File: src/phoenix/server/api/mutations/api_key_mutations.py [hunk 2]
```diff
@@ -151,10 +151,21 @@ async def create_user_api_key(
...
+    async def delete_user_api_key(
+        self, info: Info[Context, None], input: DeleteApiKeyInput
+    ) -> DeleteApiKeyMutationPayload:
+        assert (token_store := info.context.token_store) is not None
+        api_key_id = from_global_id_with_expected_type(
+            input.id, expected_type_name=UserApiKey.__name__
+        )
+        await token_store.revoke(ApiKeyId(api_key_id))
```
Comment (mikeldking) [inline src/phoenix/server/api/mutations/api_key_mutations.py:170]: Does this actually delete it from the DB?

Comment (Parker-Stafford) [reply src/phoenix/server/api/mutations/api_key_mutations.py:170]: it's tough for me to follow the inheritance but i believe so here: https://github.com/Arize-ai/phoenix/blob/auth/src/phoenix/server/jwt_store.py#L202 - this is the same logic as the system keys

File: src/phoenix/server/api/mutations/api_key_mutations.py (hunk 1 above)
Comment (Parker-Stafford) [reply src/phoenix/server/api/mutations/api_key_mutations.py:162]: oh nice this wasn't in main and misse it when i switched over

File: src/phoenix/server/api/mutations/api_key_mutations.py (hunk 2 above)
Comment (mikeldking) [reply src/phoenix/server/api/mutations/api_key_mutations.py:170]: I see, interesting.

**Review by mikeldking — APPROVED** [review]

**Pull request merged**